actioncable/lib/action_cable/connection/internal_channel.rb

```diff
@@ -31,6 +31,7 @@ def process_internal_message(message)
 
           case message['type']
           when 'disconnect'
+            transmit ActiveSupport::JSON.encode(identifier: ActionCable::INTERNAL[:identifiers][:disconnect], message: Time.now.to_i)
             logger.info "Removing connection (#{connection_identifier})"
             websocket.close
```

Comment:
I'm not certain, but there may be a timing issue with closing the connecting immediately after transmitting. Probably best to defer calling `close`.

/cc @lifo